docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+and refactored/improved.
+
+#### Key Points
+ - The yarn.lock file can be parsed with [pyarn](https://github.com/containerbuildsystem/pyarn) like in cachito
+ - Workspaces will need to be [processed separately](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L95) from package.json since they do not appear in the yarn.lock file
+ - Dev dependencies can be [determined similiarly](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L140) to cachito
+ - We will need to ensure that there are no collisions in the names of the archive files being added to the offline mirror
+
+#### PURL Examples
+[NPM PURL specification](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#npm)
+
+##### Registry Package
+```txt
+pkg:npm/@optional-scope/package-name@1.0.0
+```
+
+##### URL Package
+```txt
+pkg:npm/package-name@1.0.0?checksum=sha512:checksum&download_url=https://example.com/package-name/0101010101010101010101010101010101010101.tar.gz
+```
+
+##### Git Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101
+```
+
+##### Workspace/File Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101#subpath
+```
+
+### Avoiding Arbitrary Code Execution
+Yarn 1.x appears to honor the ```--ignore-scripts``` option for the install command. It can also be configured via environment variable.
+Used https://github.com/chmeliik/js-lifecycle-scripts/tree/yarn to verify.
+
+### Prefetch
+The offline mirror in the request output directory will be populated by running the following Yarn command in the project root:
+
+    yarn install --non-interactive --frozen-lockfile --disable-pnp --no-default-rc --ignore-engines
+
+Command: [yarn install](https://classic.yarnpkg.com/en/docs/cli/install)
+
+Options:
+ - non-interactive: Disable interactive prompts
+ - frozen-lockfile: Fail if yarn.lock updates are needed
+ - disable-pnp: Disable Plug'n'Play installation
+ - no-default-rc: Prevent Yarn from automatically detecting .yarnrc and .npmrc files
+ - ignore-engines: Ignore the engine node version during the prefetch
+
+ The behavior of Yarn can also be configured using select environment variables. The following ones should be set for the prefetch:
+
+    COREPACK_ENABLE_DOWNLOAD_PROMPT=0
+    COREPACK_ENABLE_PROJECT_SPEC=0 (If we decide to use the global default)
+    YARN_YARN_OFFLINE_MIRROR=<absolute path to the request output directory>
+    YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+    YARN_IGNORE_PATH=true (Use the corepack yarn version, not any Yarn version specified by the user)
+    YARN_IGNORE_SCRIPTS=true
```

Comment:
Where are these env vars documented? Is the official [page](https://classic.yarnpkg.com/en/docs/envvars) a joke? 
Unless I missed something this GH issue: https://github.com/yarnpkg/yarn/issues/8784 confirms what the page lists, so are we going to fall back to NPM options? I am confused, I don't dispute the variables work, it's just we need to be able to reference them to their respective docs.

Reply:
So, if an official docs doesn't exist, then each of these options, either CLI or env vars should be properly referenced by a link pointing to their code base.

Reply:
References added :+1: